src/apps/eth/eth_sign_msg.c

```diff
@@ -33,9 +41,13 @@ app_eth_sign_error_t app_eth_sign_msg(
     if (request->msg.size > 9999) {
         return APP_ETH_SIGN_ERR_INVALID_INPUT;
     }
+    // assert keypath length so that we can access the "network" piece
+    if (request->keypath_count < 2) {
+        return APP_ETH_SIGN_ERR_INVALID_INPUT;
+    }
     // Only support main net for now. Otherwise a user could be tricked into signing something for
     // main net even if they believe they are signing for testnet.
-    if (request->coin != ETHCoin_ETH) {
+    if (!_verify_network(request->coin, request->keypath[1])) {
```

Comment:
The keypath/coin check is already performed by `app_eth_address()`, so it seems unnecessary here (it would also be a duplication of `app_eth_params_get` and `rust_ethereum_keypath_is_valid_address`.

Reply:
so is it enough to call `rust_ethereum_keypath_is_address` in place of `_verify_network`?